system/Database/Forge.php

```diff
@@ -39,6 +39,7 @@ class Forge
     /**
      * List of keys.
      *
+     * @phpstan-var array|array{fields: string[], keyName: string}
```

Comment:
It seems $key is an empty array or list of array.

```suggestion
     * @phpstan-var array{}|list<array{fields: string[], keyName: string}>
```

Reply:
@phpstan-var array{}|array{array{fields: string[], keyName: string}}

Reply:
Is `array{array{fields: string[], keyName: string}}` valid type?

`list` is the type for arrays that `array_is_list()` returns true.
https://psalm.dev/docs/annotating_code/type_syntax/array_types/#lists
It is not documented in PHPStan docs, but it can be used with PHPStan.
(But PHPStan cannot yet checks if the index starts from 0).

So `array{}|list<array{fields: string[], keyName: string}>` seems good.

Actual values:
```
array (2) [
    0 => array (2) [
        'fields' => array (1) [
            0 => string (5) "email"
        ]
        'keyName' => string (0) ""
    ]
    1 => array (2) [
        'fields' => array (1) [
            0 => string (7) "country"
        ]
        'keyName' => string (0) ""
    ]
]
```

Reply:
Looks like this works